src/extension/index.ts

```diff
@@ -48,6 +53,68 @@ export async function activate(context: ExtensionContext) {
     const panel = new Panel(context, baser, store);
     disposables.push(commands.registerCommand('sarif.showPanel', () => panel.show()));
 
+    // URI handler
+    disposables.push(window.registerUriHandler({
+        async handleUri(uri: vscode.Uri) {
+            if (uri.path === '/alert') {
+                // Launched by Azure DevOps Advanced Security alert page.
+                const pairs = uri.query.split('&');
+                // Find the url parameter.
+                for (const param in pairs) {
+                    const pair = param.split('=');
+                    if (pair[0] === 'url') {
+                        // Decode the alert API URL and pass it to the load function.
+                        const url = decodeURIComponent(pair[1]);
+                        if (url.startsWith('https://advsec.dev.azure.com/')) {
+                            await loadAlertSarif(new URL(url));
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+    }));
+
+    async function loadAlertSarif(url: URL) {
+        try {
+            const session = await vscode.authentication.getSession('microsoft', ['499b84ac-1321-427f-aa17-267ca6975798/.default'], { createIfNone: true });
+            const accessToken = session?.accessToken;
```

Comment:
Should probably exit early here if they didn't sign in, rather than sending a useless HTTP request without authorization.